File: push/src/main/java/com/xiaomi/channel/commonutils/android/PermissionUtils.java
package com.xiaomi.channel.commonutils.android;

import android.content.Context;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/android/PermissionUtils.class */
public class PermissionUtils {
    public static final String readPhoneState = "android.permission.READ_PHONE_STATE";
    public static final String writeExternalStorage = "android.permission.WRITE_EXTERNAL_STORAGE";

    public static boolean checkSelfPermission(Context context, String str) {
        return context.getPackageManager().checkPermission(str, context.getPackageName()) == 0;
    }
}

File: push-legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/android/PermissionUtils.kt
package com.xiaomi.channel.commonutils.android

import android.content.Context

object PermissionUtils {
    const val readPhoneState = "android.permission.READ_PHONE_STATE"
    const val writeExternalStorage = "android.permission.WRITE_EXTERNAL_STORAGE"

    @JvmStatic
    fun checkSelfPermission(context: Context, permission: String): Boolean {
        return context.packageManager.checkPermission(permission, context.packageName) == 0
    }
}
